data_retriever.rb

```diff
@@ -0,0 +1,130 @@
+require 'nokogiri'
+require 'rubygems'
+require 'open-uri'
+require 'watir'
+require 'pry'
+require 'json'
+require_relative "Accounts.rb"
+require_relative "Transactions.rb"
+
+Selenium::WebDriver::Chrome.driver_path="/usr/local/bin/chromedriver"
+
+class Data_Retriever
+
+  def initialize()
+  #attr_reader :available_accounts, :transactions_records
+  @available_accounts = []
+  @transactions_records = []
+  @browser = Watir::Browser.new :chrome
+  end
+  #Opens the browser
+
+  def browse(url)
+    @browser.goto(url)
+    while @browser.div(:id => "oCAPICOM").exists? do sleep 1 end
+  end
+
+  #Does the authentification
+
+  def login
+    @browser.a(class: 'ng-binding').click
+    @browser.a(id: 'demo-link').click
+    @browser.element(:id => "step1").wait_until(&:present?)
+  end
+
+  #Parses
+
+  def parsing
+    #@page = Nokogiri::HTML.parse(@browser.html)
+    @page = Nokogiri.HTML(@browser.html)
+    puts "Parsing is done."
+  end
+
+  #Gathers credit card information
+

```

Comment:
you should to read ruby-style-guide.  

https://github.com/rubocop-hq/ruby-style-guide

Reply:
in progress.